Docs: Move/redirect files out of 'Create a plugin' folder

josmperez opened the pull request:
Complete the plugin-tools site reorganization project. Move/rename site, fix broken links.

Note: some links in the package were updated, so "patch" tag added

Instructions for testing redirects: docusaurus/website/README.md#Redirects

Fixes: https://github.com/grafana/plugin-tools/issues/925

Testing: Refer to [design doc](https://docs.google.com/document/d/1T3v41Ik5hOYNL0xMtPLOiK38IPGUlvsEgzFkDKvsROM/edit)

josmperez committed:
Docs: Move/redirect files out of 'Create a plugin' folder

jackw reviewed (approved):
LGTM! 🚀 

>Note: some links in the package were updated, so "patch" tag added

I can't see any changed links in package files so swapped out the patch label for docs and no-changelog. Maybe there's a commit missing here?

>Instructions for testing redirects: docusaurus/website/README.md#Redirects

I cannot see any redirects added to the docusaurus config here. Should they be included in this PR?

josmperez committed:
Check in package files and redirects

josmperez commented:
> LGTM! 🚀
> 
> > Note: some links in the package were updated, so "patch" tag added
> 
> I can't see any changed links in package files so swapped out the patch label for docs and no-changelog. Maybe there's a commit missing here?
> 
> > Instructions for testing redirects: docusaurus/website/README.md#Redirects
> 
> I cannot see any redirects added to the docusaurus config here. Should they be included in this PR?

Thanks. I forgot to check in the files outside the /docs directory. They are now checked in.

sympatheticmoose commented on docusaurus/docs/get-started/extend-configurations.md:
In the reorg proposal it was agreed to move this under Set up the development environment

josmperez replied:
changed sidebar order to 25

josmperez replied:
does this address your comment? or are you saying we agreed to create a new hierarchy level and locate these topics within it?

sympatheticmoose commented on docusaurus/docs/get-started/set-up-github-workflows.md:
As with extend config - in the reorg proposal we'd agreed to move into Set up development environment

josmperez replied:
changed sidebar order to 30

sympatheticmoose commented on docusaurus/docs/get-started/work-with-backend.md:
This was also suggested to move under Set up the development environment

josmperez replied:
changed sidebar order.

sympatheticmoose commented on docusaurus/docs/get-started/work-with-frontend.mdx:
This was also suggested to move under Set up the development environment

josmperez replied:
changed sidebar order

sympatheticmoose commented on docusaurus/docs/how-to-guides/add-return-to-previous-functionality.md:
Looking at the content - I think this is only applicable to App plugins and could therefore be moved under that section

josmperez replied:
done

sympatheticmoose commented on docusaurus/docs/get-started/get-started.mdx:
```diff
@@ -216,6 +216,6 @@ For a complete list of available CLI commands, refer to the [CLI reference](/ref
 
 - Start your plugin journey with one of our [plugin development tutorials](/tutorials/).
-- Learn how to [develop and extend](/create-a-plugin/) its functionality.
+- Learn how to [develop and extend](/how-to-guides) its functionality.
```
```suggestion
- Learn how to [extend](/how-to-guides) its functionality.
```

sympatheticmoose reviewed (changes requested):
There's a few suggestions from the reorg proposal that were made in comments and haven't been reflected here - not sure if there was a change in plan which needs to be discussed or if it was just missed? Otherwise LGTM but we need to address those comments first before merging 🙂 

Also even with the latest commit, there are no changes in the packages. I can see  (at least) references to `/extend-a-plugin` pages within:
- /packages/create-plugin/README.md
- /packages/create-plugin/src/constants.ts
- /packages/create-plugin/templates/common/.config/_eslintrc
- /packages/create-plugin/templates/common/.config/jest-setup.js
- /packages/create-plugin/templates/common/.config/jest.config.js
- /packages/create-plugin/templates/common/.config/tsconfig.json
- /packages/create-plugin/templates/common/.config/webpack/webpack.config.ts

josmperez committed:
Update docusaurus/docs/get-started/get-started.mdx

Co-authored-by: David Harris <david.harris@grafana.com>

josmperez committed:
Fixes from review

josmperez committed:
Merge remote-tracking branch 'refs/remotes/origin/josmperez/reorg-moves' into josmperez/reorg-moves

sympatheticmoose committed:
combine devenv topics

sympatheticmoose committed:
merge work with fe and be topics into get started and sdk pages

sympatheticmoose committed:
Merge branch 'main' into josmperez/reorg-moves

sympatheticmoose committed:
update create plugin links

sympatheticmoose committed:
Merge branch 'josmperez/reorg-moves' of github.com:grafana/plugin-tools into josmperez/reorg-moves

sympatheticmoose reviewed (approved):
The comments in the reorg proposal and PR review were about combining the content into other pages. As this was more complex and given time/date, I've gone ahead and made the changes directly, but I've tried to group the commits logically should we need to revert. 

The changes I have made:
1. https://github.com/grafana/plugin-tools/pull/993/commits/9c394360b519c48f024b15a87a5bb5c584744611 Took content from Setting up GitHub Workflows and Extending config into Set up your development environment. Then adjusted layout somewhat on that page and fixed up links.
1. https://github.com/grafana/plugin-tools/pull/993/commits/a3b89f2e6774e311c9c6d5792ef0771a429dec17 removed the Work with frontend/backend pages as these have generally felt confusing. I have merged this into the getting started guide and tried to tidy the workflow up generally within that page. Updating the SDK has been moved to the same page as the SDK itself. This is a relatively large commit as required a bunch of new snippets as well.
1. https://github.com/grafana/plugin-tools/pull/993/commits/bc16fb4003ef2ab1b9c0749513ae03fd8ae8a0f6 As I'd moved things around again, I updated the links you'd changed in create-plugin.

@josmperez These changes should definitely still be checked prior to merging. But the website is building without errors or missing links at least. Happy to jump on a call regarding the changes if helpful

@jackw from you I'm mostly re-asking for a review so you can see the structural changes as I'm not sure whether you'd seen in the google doc proposal.

josmperez commented:
> The comments in the reorg proposal and PR review were about combining the content into other pages. As this was more complex and given time/date, I've gone ahead and made the changes directly, but I've tried to group the commits logically should we need to revert.
> 
> The changes I have made:
> 
> 1. [9c39436](https://github.com/grafana/plugin-tools/commit/9c394360b519c48f024b15a87a5bb5c584744611) Took content from Setting up GitHub Workflows and Extending config into Set up your development environment. Then adjusted layout somewhat on that page and fixed up links.
> 2. [a3b89f2](https://github.com/grafana/plugin-tools/commit/a3b89f2e6774e311c9c6d5792ef0771a429dec17) removed the Work with frontend/backend pages as these have generally felt confusing. I have merged this into the getting started guide and tried to tidy the workflow up generally within that page. Updating the SDK has been moved to the same page as the SDK itself. This is a relatively large commit as required a bunch of new snippets as well.
> 3. [bc16fb4](https://github.com/grafana/plugin-tools/commit/bc16fb4003ef2ab1b9c0749513ae03fd8ae8a0f6) As I'd moved things around again, I updated the links you'd changed in create-plugin.
> 
> @josmperez These changes should definitely still be checked prior to merging. But the website is building without errors or missing links at least. Happy to jump on a call regarding the changes if helpful
> 
> @jackw from you I'm mostly re-asking for a review so you can see the structural changes as I'm not sure whether you'd seen in the google doc proposal.

These changes all look good so far. I just need to check the links and I'll merge later today.

sympatheticmoose commented:
Sounds good - Jack won't be able to review till next week at this stage. 

> let's give more thought to how to repurpose or retire the content later 

Just to be clear I have moved the unique content out of these pages.

jackw commented on docusaurus/docs/get-started/get-started.mdx:
````diff
@@ -169,53 +165,107 @@ The file structure should look like this:
+mage -v build:linux
+```
+
+#### Build targets
+
+| Option         | Description                                  | Example               |
+| -------------- | -------------------------------------------- | --------------------- |
+| `build:[arch]` | Builds a binary for a specific architecture. | `mage -v build:Linux` |
+
````
There is a mismatch between `amd` and `arm64` and the docker container which we should likely put a note about here. Previously we forced amd in the docker platform but that causes the backend delve dev env to fail to run on apple silicon.

Also not really sure if this should be added here but we have more than just `build:[arch]` as available targets. e.g.

```
| `build:backend` | build a production build for the current platform | `mage -v build:backend` |
```

Maybe we circle back to the whole backend build commands and expand them later on.

sympatheticmoose replied:
Given this is primarily a re-org is it ok to address this later?

jackw commented on docusaurus/docs/get-started/set-up-development-environment.mdx:
```diff
@@ -130,3 +141,252 @@ You will also notice that the `docker-compose.yaml` file also has the following
+
+### The CI workflow
+
+The CI (`ci.yml`) workflow is designed to lint, type check, and build the frontend and backend. It is also used to run tests on your plugin every time you push changes to your repository. The `create-plugin` tool helps to catch any issues early in the development process, before they become bigger problems.
```
>The `create-plugin` tool helps to catch any issues early in the development process, before they become bigger problems.

The tool scaffolds things to help catch problems however this sentence gives me the impression it is running in CI.

sympatheticmoose replied:
```suggestion
The CI (`ci.yml`) workflow is designed to lint, type check, and build the frontend and backend. It is also used to run tests on your plugin every time you push changes to your repository. Running the CI workflow helps to catch any issues early in the development process, before they become bigger problems.
```

jackw commented on docusaurus/docs/get-started/set-up-development-environment.mdx:
```diff
@@ -130,3 +141,252 @@ You will also notice that the `docker-compose.yaml` file also has the following
+
+### The CI workflow
+
+The CI (`ci.yml`) workflow is designed to lint, type check, and build the frontend and backend. It is also used to run tests on your plugin every time you push changes to your repository. The `create-plugin` tool helps to catch any issues early in the development process, before they become bigger problems.
```
It also runs the playwright-tests (`grafana/plugin-actions/e2e-version@main`) which creates overlap with this page: https://grafana-dev.com/developers/plugin-tools/e2e-test-a-plugin/ci.

sympatheticmoose replied:
the text does still say:
> It is also used to run tests

Would this address things for now and we can have more of a discussion about how to better combine the topics if required?

```suggestion
The CI (`ci.yml`) workflow is designed to lint, type check, and build the frontend and backend. It is also used to run tests on your plugin every time you push changes to your repository. The `create-plugin` tool helps to catch any issues early in the development process, before they become bigger problems. For more information on E2E testing as part of the CI workflow, refer to our [documentation](../e2e-test-a-plugin/ci.md).
```

josmperez replied:
I think so. Added this fix in this commit: https://github.com/grafana/plugin-tools/pull/993/commits/84351e9ca091635e521ab4f4f098d354951fa9a8

jackw commented on docusaurus/docs/get-started/set-up-development-environment.mdx:
````diff
@@ -130,3 +141,252 @@ You will also notice that the `docker-compose.yaml` file also has the following
 ```
 
 they are necessary to allow delve to attach to the running process and debug it and should not be used in production environments.
+
````
The following additions don't really fit with the title IMHO. 

One thing I'm constantly looking for in our docs is how to extend the create-plugin configs and I can never find them. I wouldn't expect to find them in the "setting up dev env" page.

sympatheticmoose replied:
Could you suggest an alternative - either to rename the setting up dev env page or where to have the extend config page?

josmperez replied:
> The following additions don't really fit with the title IMHO.
> 
> One thing I'm constantly looking for in our docs is how to extend the create-plugin configs and I can never find them. I wouldn't expect to find them in the "setting up dev env" page.

I have captured this issue [here](https://github.com/grafana/plugin-tools/issues/1013) so it doesn't delay the re-org.

jackw commented on docusaurus/docs/troubleshooting.md:
```diff
@@ -62,7 +62,7 @@ If you're using mounts to access your files, it's likely that webpack isn't dete
 
 - If you're editing your code from a native Windows app (such as VS Code), you need to manually rerun `yarn build` every time you want to see a change in your plugin.
-- Use [webpack `watchOption` with `poll`](https://webpack.js.org/configuration/watch/#watchoptionspoll) in your project. Find instructions on how to extend webpack configuration in our [documentation](https://grafana.com/developers/plugin-tools/create-a-plugin/extend-configurations).
+- Use [webpack `watchOption` with `poll`](https://webpack.js.org/configuration/watch/#watchoptionspoll) in your project. Find instructions on how to extend webpack configuration in our [documentation](./get-started/set-up-development-environment.mdx#extend-configurations).
```
This has been built into the [create-plugin webpack](https://github.com/grafana/plugin-tools/blob/main/packages/create-plugin/templates/common/.config/webpack/webpack.config.ts#L241-L246) configs for ~10 months now. I would suggest asking devs to run the `create-plugin update` command over suggesting to customise the config.

sympatheticmoose replied:
```suggestion
- Ensure you are running the latest recommended webpack configuration by running the [update](./migration-guides/update-create-plugin-versions.mdx) command in `create-plugin`.
```
How's this

josmperez replied:
works for me.

sympatheticmoose commented on docusaurus/docs/snippets/createplugin-build-fe-prod.npm.md:
````diff
@@ -0,0 +1,3 @@
+```shell
+npm run prod
````
🤦 terrible mistake, glad you caught it!

sympatheticmoose committed:
fix npm build cmd

josmperez committed:
Update docusaurus/docs/get-started/set-up-development-environment.mdx

Co-authored-by: David Harris <david.harris@grafana.com>

josmperez committed:
Fix to dev env setup page

josmperez commented:
Thanks for your work on this topic, getting it across the finish line @sympatheticmoose @jackw @academo ! One small step for plugin-tools, one giant leap for mankind!
